homeassistant/components/switch/rainmachine.py

```diff
@@ -64,46 +67,48 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     password = config.get(CONF_PASSWORD)
     _LOGGER.debug('Password: %s', password)
 
-    hide_disabled_entities = config.get(CONF_HIDE_DISABLED_ENTITIES)
-    _LOGGER.debug('Show disabled entities: %s', hide_disabled_entities)
-
     zone_run_time = config.get(CONF_ZONE_RUN_TIME)
     _LOGGER.debug('Zone run time: %s', zone_run_time)
 
     try:
         if ip_address:
-            _LOGGER.debug('Configuring local API...')
-            auth = rm.Authenticator.create_local(ip_address, password)
+            port = config.get(CONF_PORT)
+            _LOGGER.debug('Port: %s', port)
+
+            ssl = config.get(CONF_SSL)
+            _LOGGER.debug('SSL: %s', ssl)
+
+            _LOGGER.debug('Configuring local API')
+            auth = rm.Authenticator.create_local(
+                ip_address, password, port=port, https=ssl)
         elif email_address:
-            _LOGGER.debug('Configuring remote API...')
+            _LOGGER.debug('Configuring remote API')
             auth = rm.Authenticator.create_remote(email_address, password)
 
-        _LOGGER.debug('Instantiating RainMachine client...')
+        _LOGGER.debug('Querying against: %s', auth.url)
+
+        _LOGGER.debug('Instantiating RainMachine client')
         client = rm.Client(auth)
 
         rainmachine_device_name = client.provision.device_name().get('name')
 
         entities = []
         for program in client.programs.all().get('programs'):
-            if hide_disabled_entities and program.get('active') is False:
-                continue
-
-            _LOGGER.debug('Adding program: %s', program)
-            entities.append(
-                RainMachineProgram(
-                    client, program, device_name=rainmachine_device_name))
+            if program.get('active'):
```

Comment:
I was referring to using a guard clause to let the normal path of execution be not indented. But there's no nesting here so it doesn't matter really.